ogr/ogrsf_frmts/gpkg/ogrgeopackagedatasource.cpp

```diff
@@ -6005,6 +6005,21 @@ GDALDataset *GDALGeoPackageDataset::CreateCopy(const char *pszFilename,
                                                GDALProgressFunc pfnProgress,
                                                void *pProgressData)
 {
+    const int nBands = poSrcDS->GetRasterCount();
+    if (nBands == 0)
+    {
+        GDALDataset *poDS = nullptr;
+        GDALDriver *poThisDriver =
+            GDALDriver::FromHandle(GDALGetDriverByName("GPKG"));
+        if (poThisDriver != nullptr)
```

Comment:
can this actually happen?

Reply:
Really unlikely. That would be an equivalent of the [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) pattern for GDAL, where one thread would CreateCopy() whereas another one would deregister the GPKG driver...